cmd/upload.go

```diff
@@ -210,6 +218,147 @@ func genStats(durations []float64) stats {
 	}
 }
 
+type uploadProgressBar struct {
+	progress           *mpb.Progress
+	bar                *mpb.Bar
+	progressTotal      int64
+	progressTotalMutex *sync.Mutex
+}
+
+func newUploadProgressBar() *uploadProgressBar {
+	progress := mpb.New()
+	var progressTotal int64
+	bar := progress.AddBar(progressTotal,
+		mpb.BarRemoveOnComplete(),
+		mpb.PrependDecorators(
+			decor.Name("upload", decor.WC{W: 7, C: decor.DidentRight}),
+			decor.OnComplete(decor.EwmaETA(decor.ET_STYLE_GO, 60, decor.WC{W: 4}), "done"),
+		),
+		mpb.AppendDecorators(decor.Percentage()),
+	)
+
+	return &uploadProgressBar{
+		progress:           progress,
+		bar:                bar,
+		progressTotal:      progressTotal,
+		progressTotalMutex: &sync.Mutex{},
+	}
+}
+
+func (upb *uploadProgressBar) incrementTotal() {
+	upb.progressTotalMutex.Lock()
+	upb.progressTotal += int64(1)
+	upb.bar.SetTotal(upb.progressTotal, false)
+	upb.progressTotalMutex.Unlock()
+}
+
+func (upb *uploadProgressBar) increment(wdd ...time.Duration) {
+	upb.bar.Increment(wdd...)
+}
+
+func (upb *uploadProgressBar) wait() {
+	upb.progress.Wait()
+}
+
+type processableFiles struct {
+	singleBundleFiles []string
+	multiBundleFiles  []string
+}
+
+func filterProcessableFiles(dir string) (processableFiles, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return processableFiles{}, err
+	}
+
+	var procFiles processableFiles
+
+	for _, file := range files {
+		if !file.IsDir() {
+			if strings.HasSuffix(file.Name(), ".json") {
+				procFiles.singleBundleFiles = append(procFiles.singleBundleFiles, filepath.Join(dir, file.Name()))
+			} else if strings.HasSuffix(file.Name(), ".ndjson") {
+				procFiles.multiBundleFiles = append(procFiles.multiBundleFiles, filepath.Join(dir, file.Name()))
+			}
+		}
+	}
+
+	return procFiles, nil
+}
+
+func enqueueBundlesFromSingleBundleFiles(files []string, uploadQueue chan<- bundle, wg *sync.WaitGroup, progressBar *uploadProgressBar) {
+	for _, file := range files {
+		progressBar.incrementTotal()
```

Comment:
Yeah, I was under the misconception that the total value (capacity of the progress bar) should be adjusted as soon as a bundle is read. The idea was that this immediately reflects in the progress bar being rendered. But since the channel itself has a capacity it doesn't make a difference whether it is done within the producer or the consumer.

However, I would still keep the progress bar wrapper for a more convenient adjustment of the total value.

So, will be adjusted, yes.

Reply:
Ok thanks.